libc/dirent/rewinddir.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * test/test_strftime.c
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Arkadiusz Kozlowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <time.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <dirent.h>
+#include <limits.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <string.h>
+
+#include <unity_fixture.h>
+
+#include "common.h"
+
+#define MAIN_DIR "test_rewinddir"
+
+
+TEST_GROUP(dirent_rewinddir);
+
+TEST_SETUP(dirent_rewinddir)
+{
+	mkdir(MAIN_DIR, 0777);
+	mkdir(MAIN_DIR "/dir1", 0777);
+	mkdir(MAIN_DIR "/dir2", 0777);
```

Comment:
```suggestion
	mkdir(MAIN_DIR, S_IRUSR);
	mkdir(MAIN_DIR "/dir1", S_IRUSR);
	mkdir(MAIN_DIR "/dir2", S_IRUSR);
```
S_IRUSR is sufficient for reading with `readdir()`